backend/src/plugins/Automod/actions/setSlowmode.ts

```diff
@@ -0,0 +1,52 @@
+import * as t from "io-ts";
+import { automodAction } from "../helpers";
+import { convertDelayStringToMS, isDiscordRESTError, tDelayString, tNullable } from "../../../utils";
+import { LogType } from "../../../data/LogType";
+import { AnyGuildChannel } from "eris";
+
+export const SetSlowmodeAction = automodAction({
+  configType: t.type({
+    channels: t.array(t.string),
+    duration: tNullable(tDelayString),
+  }),
+
+  defaultConfig: {
+    duration: "10s",
+  },
+
+  async apply({ pluginData, actionConfig }) {
+    const duration = actionConfig.duration ? convertDelayStringToMS(actionConfig.duration)! : 0;
+
+    for (const channelId of actionConfig.channels) {
+      const channel = pluginData.guild.channels.get(channelId);
+      // 2 = Guild Voice, 5 = Guild News - Both dont allow slowmode
+      if (!channel || channel.type === 2 || channel.type === 5) continue;
```

Comment:
Stage channels probably don't either. Could filter to allowed channel types rather than skipping specific ones?

Reply:
Could also use `Constants.ChannelTypes` from eris. See: https://github.com/abalabahaha/eris/blob/master/lib/Constants.js#L230